### ISSUE TITLE
bug: Long texts in ion-checkbox

### ISSUE THREAD

**Issue by Marius-Romanus** [ionic-team/ionic-framework#28919]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

v7.x

### Current Behavior

Hello again!

I notice strange things with the checkboxes.

1. Within an ion-item and an ion-checkbox. By having a long text, and an icon as a slot end, the icon moves down instead of staying to the right as before.
2. It seems strange to me to have to put a span or an ion-text because in ion-checkbox with very long texts the ion-text-wrap class does not affect it. (This may be so and that's it).

<img width="1268" alt="Captura de pantalla 2024-01-30 140344" src="https://github.com/ionic-team/ionic-framework/assets/18282247/4e6ab50b-7592-4fd4-8953-b7965d73ad55">

3. When you need an ion-input inside an ion-checkbox, is the checkbox syntax still the old syntax?

  I understand from this link that yes, https://ionicframework.com/docs/api/item#controls
  
  But the css alignment of the different elements is not the same in the old way and the new way. So if you have a list with a mix of labels and inputs, it will not look aligned.

<img width="295" alt="Captura de pantalla 2024-01-30 140259" src="https://github.com/ionic-team/ionic-framework/assets/18282247/bef8de66-3ba0-43d7-b8b2-96a3776c0fc4">

greetings!

### Expected Behavior

1. The final icons within an ion-input should not go to the bottom.
2. Checkbox labels should be able to support long texts.
3. Checkboxes with labels or inputs should be aligned (I have only tested in MD)

### Steps to Reproduce

I leave you a repo where the code for the screenshots is. It's all in src\app\tab1\tab1.page.html

### Code Reproduction URL

https://github.com/Marius-Romanus/ionic-checkbox

### Ionic Info

Ionic:

Ionic CLI : 7.2.0
Ionic Framework : @ionic/angular 7.6.6
@angular-devkit/build-angular : 17.1.1
@angular-devkit/schematics : 17.1.1
@angular/cli : 17.1.1
@ionic/angular-toolkit : 11.0.0

Capacitor:

Capacitor CLI : 5.6.0
@capacitor/android : not installed
@capacitor/core : 5.6.0
@capacitor/ios : not installed

Utility:

cordova-res : not installed globally
native-run : 2.0.1

System:

NodeJS : v20.9.0
npm : 10.1.0
OS : Windows 10

**Comment by brandyscarney** [comment]
Thank you for reporting this issue! This bug has been resolved in version [7.6.7](https://github.com/ionic-team/ionic-framework/releases/tag/v7.6.7). Here is a comparison of your reported issue in version `7.6.6` and the fixed version, `7.6.7`:

- [Ionic 7.6.6 Codepen](https://codepen.io/brandyscarney/pen/VwRQJKP)
- [Ionic 7.6.7 Codepen](https://codepen.io/brandyscarney/pen/vYPdqyZ)

Please update to the latest version of `@ionic/angular` to resolve this. If you are still seeing problems, please open a new issue. Thank you! 🙂

**Comment by Marius-Romanus** [comment]
Hello @brandyscarney, in the ionic 7 documentation you indicate that you have to use this new way, even though the old one can continue to be used. But in later versions it will not be possible. But you really can't do the same thing, or it doesn't appear in the ion-checkbox documentation, for example before you could do this and now you can't.


<img width="563" alt="image" src="https://github.com/ionic-team/ionic-framework/assets/18282247/8e101598-5e0e-4e1a-a0b2-52113c217126">



```
      <ion-item>
        <ion-checkbox slot="start"></ion-checkbox>
        <ion-avatar slot="start">
          <img src="https://ionicframework.com/favicon-96x96.png">
        </ion-avatar>
        <ion-label>
          <h2 class="ion-text-wrap">Lorem ipsum dolor sit amet consectetur adipisicing elit.</h2>
          <h3 class="ion-text-wrap"> Iste aperiam, ut dolore amet architecto unde officia in autem quas nihil itaque ratione eveniet.</h3>
        </ion-label>
      </ion-item>
```
